lib/routes/routeBackbeat.js

```diff
@@ -472,14 +472,29 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
                 omVal[headerName] = objMd[headerName];
             });
         }
+
         const versionId = decodeVersionId(request.query);
+        const versioning = bucketInfo.isVersioningEnabled();
+
         // specify both 'versioning' and 'versionId' to create a "new"
         // version (updating master as well) but with specified
         // versionId
         const options = {
-            versioning: bucketInfo.isVersioningEnabled(),
+            versioning,
```

Comment:
It could be cleaner to
1. prepare the final `versioning` and `versionId` fields to pass in options
2. build the `options` object with those two fields

Reply:
I did not change the existing logic much to have the change easily forwarded to development/8.x branches.